spec/issues/591_delete_method_regression_spec.rb

```diff
@@ -0,0 +1,53 @@
+require 'spec_helper'
+
+describe '#591 delete method regression' do
+  include_context "#{MODEL_PARSER} swagger example"
+
+  let(:app) do
+    class Foo < Grape::Entity
+      expose :something
+    end
+
+    Class.new(Grape::API) do
+      namespace :issue_591 do
+        desc 'delete with model, code 200',
+             is_array: false,
+             success: { code: 200, model: Foo, message: 'deleted' }
+        delete '/deleteme200' do
+          present({ something: 'bar' }, with: Foo)
+        end
+
+        desc 'delete with model, code 204',
+             is_array: false,
+             success: { code: 204, model: Foo, message: 'deleted' }
```

Comment:
Correct.  The existing code (prior to this change) modified the 204 to be a 200 if a response body (model) was given.  I maintained that behavior here (ie, I'm not introducing something new here).